psychopy_linux_installer

```diff
@@ -123,6 +133,72 @@ log() {
     fi
 }
 
+sudo_wrapper() {
+    if [[ "$sudo_mode" == "force" ]]; then
+        log sudo "$@"
+    else
+        error_output=$("$@" 2>&1)
+        exit_code=$?
+
+        case "$sudo_mode" in
+            ask)
+                # Determine if the command is a package manager command
+                if [[ "$1" == "$pkg_manager" ]]; then
+                    is_pkg_manager_command=true
+                else
+                    is_pkg_manager_command=false
+                fi
+
+                # Handle package manager commands
+                if [ $exit_code -eq 126 ] || [ $exit_code -eq 100 ] || 
+                ([ $exit_code -ne 0 ] && log_message "$error_output" | grep -Eqi "permission denied|requires root|must be root|are not authorized|could not open|operation not permitted|cannot open|superuser privileges|has to be run with"); then
```

Comment:
FWIW, I prefer avoiding duplication.  why not to define some helper `sudo_is_needed` with this code which would replace here and below just to

```suggestion
                if sudo_is_needed "$exit_code" "$error_output"; then
```
?